falkordb/falkordb.py

```
@@ -225,6 +227,36 @@
+            ValueError: If the `mode` is neither Sentinel nor Cluster.
+        """
+        # decide if it's Sentinel or cluster
+        mode = self.connection.execute_command("info")['redis_mode']
+        if hasattr(self, 'sentinel') and self.sentinel is not None:
+            replica_hostnames = self.sentinel.discover_slaves(service_name=self.service_name)
+            return [(host, port) for host, port in replica_hostnames]
+        elif mode == "cluster":
+            data = self.connection.cluster_nodes()
+            return [(flag['hostname'], ip_port.split(':')[1]) for ip_port, flag in data.items() if 'slave' in flag["flags"]]
+        else:
+            raise ValueError(f"Unsupported Redis mode: {redis_mode}")
```

Ruff

`redis_mode` may be undefined, or defined from star imports

(F405)